lib/msf/core/handler/reverse_tcp.rb

```diff
@@ -89,7 +89,12 @@ def human_name
   def payload_uri
     addr = datastore['LHOST']
     uri_host = Rex::Socket.is_ipv6?(addr) ? "[#{addr}]" : addr
-    "tcp://#{uri_host}:#{datastore['LPORT']}"
+    if listener_sock.nil?
+      via = "(setting up)"
+    else
+      via = via_string(listener_sock.client) if listener_sock.respond_to?(:client)
+    end
+    "tcp://#{uri_host}:#{datastore['LPORT']} #{via}"
```

Comment:
I like the intent of the PR, I'm not sure if we want to be modifying this method to no longer return a uri, as it might break assumptions elsewhere in the code. Would it be possible to add an alternative method that adds this additional human readable metadata to the console? Or maybe this logic should live in the calling logic if that's an easier approach

Reply:
I've now instead modified the calling logic.